### DISCUSSION TITLE
Tm1 to Python

### DISCUSSION THREAD

**Discussion by rdb008** [cubewise-code/tm1py#849] [Q&A]
Hi,

we use TM1 mostly through Excel Add-on and I am trying to get the date through Python.

I am using the basic authentication method as mentioned here (https://www.ibm.com/docs/en/cloud-paks/cp-data/4.0?topic=notebook-adding-data-from-planning-analytics-connection) - not sure if it is OK to do this but i believe this is how Excel is also connecting.

when i run the code:

A1 = 'https://XXXXXX.ibm.com'
A3 = 'USER@COMPANY.COM'
A4 = 'PASSWORD'

tm1 = TM1Service(base_url=A1, user=A3, password=A4, ssl=False)

i am receiving the below error:

TM1pyRestException: Text: '{ "error":"access_denied", "error_description":"Failed to verify OAuth information. Transaction ID=XXX" }' - Status Code: 401 - Reason: 'Unauthorized' - Headers: {'Date': 'XXXXXX', 'Server': '', 'Strict-Transport-Security': 'max-age=XXX; includeSubdomains', 'Content-Type': 'application/json', 'WWW-Authenticate': 'Bearer', 'X-Backside-Transport': 'FAIL FAIL', 'Vary': 'Accept-Encoding', 'Content-Encoding': 'gzip', 'X-Content-Type-Options': 'nosniff', 'Content-Length': '120', 'Keep-Alive': 'timeout=5, max=100', 'Connection': 'Keep-Alive'}

any help is welcome!!

**Comment by MariusWirtz** [comment]
Hi @rdb008,

I hope my answer in the tm1forum helped. Were you able to resolve the issue?
https://www.tm1forum.com/viewtopic.php?t=16398
